pkg/detectors/youtubeapikey/youtubeapikey.go

```diff
@@ -42,36 +45,111 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
-						s1.Verified = true
+		if verify {
+			// Try verification with channel IDs if available
+			var isVerified bool
+			var verificationErr error
+
+			if len(idmatches) > 0 {
+				// Try with available channel IDs first
+				for _, idmatch := range idmatches {
+					resIdmatch := strings.TrimSpace(idmatch[1])
+					isVerified, verificationErr = verifyWithChannelId(ctx, client, resMatch, resIdmatch)
+					if isVerified {
+						break
 					}
 				}
 			}
 
-			results = append(results, s1)
+			// If no channel IDs or verification failed, try simpler verification
+			if !isVerified {
+				isVerified, verificationErr = verifyApiKey(ctx, client, resMatch)
+			}
+
+			s1.Verified = isVerified
+			s1.SetVerificationError(verificationErr, resMatch)
 		}
 
+		results = append(results, s1)
 	}
 
 	return results, nil
 }
 
+func verifyWithChannelId(ctx context.Context, client *http.Client, apiKey, channelId string) (bool, error) {
+	// Reference: https://developers.google.com/youtube/v3/docs/channelSections/list
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet,
+		fmt.Sprintf("https://www.googleapis.com/youtube/v3/channelSections?key=%s&channelId=%s&part=snippet", apiKey, channelId),
+		http.NoBody)
+	if err != nil {
+		return false, err
+	}
+
+	return executeRequest(client, req)
+}
+
+func verifyApiKey(ctx context.Context, client *http.Client, apiKey string) (bool, error) {
+	// Use a simpler endpoint that doesn't require specific IDs
+	// Reference: https://developers.google.com/youtube/v3/docs/i18nLanguages/list
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet,
+		fmt.Sprintf("https://www.googleapis.com/youtube/v3/i18nLanguages?key=%s&part=snippet", apiKey),
+		http.NoBody)
+	if err != nil {
+		return false, err
+	}
+
+	return executeRequest(client, req)
+}
+
+func executeRequest(client *http.Client, req *http.Request) (bool, error) {
+	res, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+	defer func() {
+		_, _ = io.Copy(io.Discard, res.Body)
+		_ = res.Body.Close()
+	}()
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		return true, nil
+	case http.StatusBadRequest:
+		// 400: Could be invalid API key or other parameter issues
+		body, _ := io.ReadAll(res.Body)
+		if strings.Contains(string(body), "keyInvalid") {
+			return false, fmt.Errorf("invalid API key")
+		}
+		if strings.Contains(string(body), "quotaExceeded") {
+			return true, fmt.Errorf("quota exceeded (valid key): %d", res.StatusCode)
+		}
+		return false, nil
+	case http.StatusUnauthorized:
+		// 401: Missing or invalid authentication credentials
+		return false, fmt.Errorf("unauthorized: %d", res.StatusCode)
+	case http.StatusForbidden:
+		// 403: API key is valid but access is forbidden (quota exceeded, API not enabled, etc.)
+		body, _ := io.ReadAll(res.Body)
+		if strings.Contains(string(body), "quotaExceeded") {
+			return true, fmt.Errorf("quota exceeded (valid key): %d", res.StatusCode)
```

Comment:
If we're returning true, that means the key was successfully verified. So we shouldn't return a verification error in that case.